src/ring_theory/localization.lean

```diff
@@ -1009,9 +1009,20 @@ lemma coe_submodule_mono {I J : ideal R} (h : I ≤ J) :
   coe_submodule S I ≤ coe_submodule S J :=
 submodule.map_mono h
 
+@[simp] lemma coe_submodule_bot : coe_submodule S (⊥ : ideal R) = ⊥ :=
+by rw [coe_submodule, submodule.map_bot]
+
 @[simp] lemma coe_submodule_top : coe_submodule S (⊤ : ideal R) = 1 :=
 by rw [coe_submodule, submodule.map_top, submodule.one_eq_range]
 
+@[simp] lemma coe_submodule_sup (I J : ideal R) :
+  coe_submodule S (I ⊔ J) = coe_submodule S I ⊔ coe_submodule S J :=
```

Comment:
`coe_ideal_sup` uses `I + J`, this uses `I ⊔ J` - which is the simp-normal form?

Reply:
For submodules, the simp-normal form is `⊔` (what you see here), for fractional ideals the normal form is `+` (what you see in `coe_ideal_sup`); both are defeq. In maths `+` really is the standard notation for this operation but the convention for submodules seems to have shifted towards `⊔`, presumably because we needed more order lemmas than ring lemmas at some point. I don't think that the convenient automation weighs up to the unfamiliarity, and especially what I'm working with is more of a ring structure than an order structure, so I'm not going to change the convention for fractional ideals unless there is a really convincing reason.

In particular I'm wondering why we don't just use a `reducible` instance for `has_add` on submodules, then `simp` should be able to see through the `+` to apply `⊔`-lemmas.

Reply:
My point is simply that `I ⊔ J` appears on the LHS of this lemma but `I + J` appears on the right above. As I read it, the expression is of type `ideal` in both cases, but maybe I'm missing a coercion. If there's no coercion in either case, then shouldn't we be consistent?

Reply:
Also, I don't think the type of reducibility you're asking for can exist.

Reply:
> My point is simply that `I ⊔ J` appears on the LHS of this lemma but `I + J` appears on the right above. As I read it, the expression is of type `ideal` in both cases, but maybe I'm missing a coercion. If there's no coercion in either case, then shouldn't we be consistent?

I'm very confused about this question, these are different operations with different names and different types :S
 * `coe_submodule S : ideal R → submodule R S`, where `algebra R S`
 * `coe : ideal R → fractional_ideal M Rₘ`, where `is_localization M Rₘ`
It happens to be the case that coercing (integral) ideals to fractional ideals is defined in terms of `coe_submodule`, but that is just an implementation detail.

I guess I could add explicit `↑`s to every term on the RHS, but that doesn't seem to be the convention at all in other parts of mathlib.